lib/hammer_cli_foreman/api/connection.rb

```diff
@@ -1,22 +1,35 @@
 require 'hammer_cli_foreman/api/session_authenticator_wrapper'
+require 'hammer_cli_foreman/api/authenticator'
 require 'hammer_cli_foreman/api/interactive_basic_auth'
+require 'hammer_cli_foreman/api/oauth/interactive_keycloak_auth_via_2fa'
+require 'hammer_cli_foreman/api/oauth/interactive_keycloak_auth'
 require 'hammer_cli_foreman/api/void_auth'
+require 'uri'
 
 module HammerCLIForeman
+  AUTH_TYPES = {
+    basic_auth: "Basic_Auth",
```

Comment:
We should have documented those in `foreman.yml` so users know what exactly they should specify.